src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -151,36 +156,44 @@ private Task OnProjectFactoryCompletedAsync()
             AggregateWorkspaceProjectContext previousContextToDispose = null;
             using (await _gate.DisposableWaitAsync().ConfigureAwait(false))
             {
+                string newTargetFramework = null;
+                var projectProperties = await _commonServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
+
                 // Check if we have already computed the project context.
                 if (_currentAggregateProjectContext != null)
                 {
-                    // For non-cross targeting projects, we can use the current project context.
-                    // For cross-targeting projects, we need to verify that the TargetFrameworks for the current project context matches latest TargetFrameworks project property value.
+                    // For non-cross targeting projects, we can use the current project context if the TargetFramework hasn't changed.
+                    // For cross-targeting projects, we need to verify that the current project context matches latest frameworks targeted by the project.
                     // If not, we create a new one and dispose the current one.
 
                     if (!_currentAggregateProjectContext.IsCrossTargeting)
                     {
-                        return _currentAggregateProjectContext;
-                    }
-
-                    var projectProperties = await _commonServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
-                    var targetFrameworks = (string)await projectProperties.TargetFrameworks.GetValueAsync().ConfigureAwait(false);
+                        newTargetFramework = (string)await projectProperties.TargetFramework.GetValueAsync().ConfigureAwait(false);
```

Comment:
AggregateWorkspaceProjectContext.IsCrossTargeting will be true for projects based on TargetFrameworks, so the if condition is only satisified for TargetFramework based project.

More generally, these issues don't repro for TargetFrameworks based cross targeting projects, as the project configuration itself has a "TargetFramework" dimension for each TF in TargetFrameworks property. This is turn means we have separate configured projects per TF, and changing this property causes a new configured project to be loaded. However, for TargetFramework based project, we don't have a separate TargetFramework project configuration dimension, so we are essentially using the same configured project when the property is edited, hence these intellisense issues.